src/solvers/refinement/string_constraint_generator_insert.cpp

```diff
@@ -74,8 +74,9 @@ exprt string_constraint_generatort::add_axioms_for_insert(
   PRECONDITION(f.arguments().size() == 5 || f.arguments().size() == 7);
   array_string_exprt s1 = get_string_expr(f.arguments()[2]);
   array_string_exprt s2 = get_string_expr(f.arguments()[4]);
-  array_string_exprt res =
-    char_array_of_pointer(f.arguments()[1], f.arguments()[0]);
+  const array_string_exprt res =
+    array_pool.find(f.arguments()[1], f.arguments()[0]);
+  // \todo: is this really 3 and not 4 ?
   const exprt &offset = f.arguments()[3];
```

Comment:
Isn't there already an array at index 4? Having both offset and array at that place would seem surprising?

Reply:
I was thinking the two were swapped because in the documentation of the function the order is different, but actually it's the documentation that needs to be updated.